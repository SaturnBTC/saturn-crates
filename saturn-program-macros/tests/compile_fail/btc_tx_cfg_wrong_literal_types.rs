use saturn_program_macros::saturn_program;

mod instruction {
    #[derive(borsh::BorshSerialize, borsh::BorshDeserialize)]
    pub enum Instr {
        Dummy,
    }
}

#[saturn_program(
    instruction = "crate::instruction::Instr",
    btc_tx_cfg(max_inputs_to_sign = "four", max_modified_accounts = true)
)]
mod handlers {}

fn main() {} 