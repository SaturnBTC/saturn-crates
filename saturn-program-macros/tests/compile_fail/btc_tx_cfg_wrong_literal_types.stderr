error: max_inputs_to_sign must be an integer literal
  --> tests/compile_fail/btc_tx_cfg_wrong_literal_types.rs:13:37
   |
13 |     btc_tx_cfg(max_inputs_to_sign = "four", max_modified_accounts = true)
   |                                     ^^^^^^
